utility/daemon/daemon.js

```diff
@@ -32,16 +33,30 @@ module.exports = function(RED) {
 
         function inputlistener(msg) {
             if (msg != null) {
-                if (msg.hasOwnProperty("kill") && node.running) {
+                if (msg.hasOwnProperty("stop")) {
+                    node.stopped = true;
+                    if (node.running) {
+                        node.child.kill(node.closer);
+                    }
+                    node.debug(node.cmd+" stopped by msg");
+                    node.status({fill:"grey",shape:"ring",text:RED._("daemon.status.stopped")});
+                }
+                else if (msg.hasOwnProperty("kill") && node.running) {
                     if (typeof msg.kill !== "string" || msg.kill.length === 0 || !msg.kill.toUpperCase().startsWith("SIG") ) { msg.kill = "SIGINT"; }
                     node.child.kill(msg.kill.toUpperCase());
                 }
-                else if (msg.hasOwnProperty("start") && !node.running) {
-                    let args = "";
-                    if (msg.hasOwnProperty("args") && msg.args.length > 0) {
-                        args = parseArgs(msg.args.trim());
+                else if (msg.hasOwnProperty("start")) {
+                    node.stopped = false;
+                    if (!node.running) {
+                        let args = "";
+                        if (msg.hasOwnProperty("args") && msg.args.length > 0) {
+                            args = parseArgs(msg.args.trim());
+                        }
+                        runit(args);
+                    } else {
```

Comment:
Why are we doing this here ? If it is called with start and is already running then we don't want to kill it do we ? Surely just leave it running.